datasetinsights/io/download.py

```diff
@@ -122,3 +126,13 @@ def _crc32_checksum(filepath):
         checksum = zlib.crc32(f.read())
 
     return checksum
+
+
+def _md5_checksum(filename):
+    """ Calculate the checksum of a file using MD5.
+    """
+    md5 = hashlib.md5()
+    with open(filename, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
+            md5.update(chunk)
+    return base64.b64encode(md5.digest()).decode("utf-8")
```

Comment:
Why is this checksum base64 encoded and decoded to "utf-8"? Why we can't use the md5 checksum as is?

Reply:
GCS encode MD5 hash of the data; encoded using base64

Ref: https://cloud.google.com/storage/docs/json_api/v1/objects